### DISCUSSION TITLE
Can extensions contribute to editor tab context menu?

### DISCUSSION THREAD

**Discussion by gjsjohnmurray** [microsoft/vscode-discussions#852] [Extension Development QnA] [answered]
This section in the [reference docs](https://code.visualstudio.com/api/references/contribution-points#contributes.menus) hints that it should be possible for extensions to add items to the editor tab context menu:

![image](https://github.com/microsoft/vscode-discussions/assets/6726799/7bbfdd24-f72f-47f1-914e-e14a1af4cceb)

But I haven't found which `contributes.menus` entry we should be using. Does anyone know?

**Comment by alefragnani** [comment, marked as answer]
Hi @gjsjohnmurray ,

I guess you are looking for `"editor/title/context"`.

From my Bookmarks extension, changed the entry  `"editor/title` (which is that ellipsis button in the tab - title area) to `"editor/title/context"`

```json
"editor/title/context": [
                {
                    "submenu": "bookmarks.editor.title",
                    "group": "bookmarks"
                }
            ],
```

Where `bookmarks.editor.title` is: 

```json
"bookmarks.editor.context": [
                {
                    "command": "bookmarks.toggle",
                    "group": "bookmarks",
                    "when": "editorTextFocus && config.bookmarks.showCommandsInContextMenu"
                },
                {
                    "command": "bookmarks.toggleLabeled",
                    "group": "bookmarks",
                    "when": "editorTextFocus && config.bookmarks.showCommandsInContextMenu"
                },
                {
                    "command": "bookmarks.jumpToNext",
                    "group": "bookmarks@1",
                    "when": "editorTextFocus && config.bookmarks.showCommandsInContextMenu"
                },
                {
                    "command": "bookmarks.jumpToPrevious",
                    "group": "bookmarks@1",
                    "when": "editorTextFocus && config.bookmarks.showCommandsInContextMenu"
                }
            ],
```

Gave me this:

![image](https://github.com/microsoft/vscode-discussions/assets/3781424/b655d69c-de99-4ec0-9599-92b65198481e)

Hope this helps

**Reply by gjsjohnmurray** [reply]
Thanks @alefragnani
